packages/tokens/fungible/src/extensions/allowlist/storage.rs

```diff
@@ -0,0 +1,212 @@
+    /// * data - `[]`
+    ///
+    /// # Security Warning
+    ///
+    /// **IMPORTANT**: This function bypasses authorization checks and should
+    /// only be used:
+    /// - During contract initialization/construction
+    /// - In admin functions that implement their own authorization logic
+    ///
+    /// Using this function in public-facing methods creates significant
+    /// security risks as it could allow unauthorized allowlist
+    /// modifications.
+    pub fn allow_user(e: &Env, user: &Address) {
+        // Set the user as allowed
+        let key = AllowListStorageKey::Allowed(user.clone());
+        e.storage().persistent().set(&key, &true);
+        e.storage().persistent().extend_ttl(&key, BALANCE_TTL_THRESHOLD, BALANCE_EXTEND_AMOUNT);
+
+        // Emit event
+        emit_user_allowed(e, user);
+    }
+
+    /// Disallows a user from receiving and transferring tokens.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `user` - The address to disallow.
+    ///
+    /// # Events
+    ///
+    /// * topics - `["user_disallowed", user: Address]`
+    /// * data - `[]`
+    ///
+    /// # Security Warning
+    ///
+    /// **IMPORTANT**: This function bypasses authorization checks and should
+    /// only be used:
+    /// - During contract initialization/construction
+    /// - In admin functions that implement their own authorization logic
+    ///
+    /// Using this function in public-facing methods creates significant
+    /// security risks as it could allow unauthorized allowlist
+    /// modifications.
+    pub fn disallow_user(e: &Env, user: &Address) {
+        // Set the user as not allowed
+        let key = AllowListStorageKey::Allowed(user.clone());
+        e.storage().persistent().set(&key, &false);
+        e.storage().persistent().extend_ttl(&key, BALANCE_TTL_THRESHOLD, BALANCE_EXTEND_AMOUNT);
+
+        // Emit event
+        emit_user_disallowed(e, user);
+    }
+
+    // ################## OVERRIDDEN FUNCTIONS ##################
+
+    /// Transfers `amount` of tokens from `from` to `to`.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to Soroban environment.
+    /// * `from` - The address holding the tokens.
+    /// * `to` - The address receiving the transferred tokens.
+    /// * `amount` - The amount of tokens to be transferred.
+    ///
+    /// # Errors
+    ///
+    /// * [`FungibleTokenError::UserNotAllowed`] - When either `from` or `to` is
+    ///   not allowed.
+    /// * Also refer to [`Base::transfer`] errors.
+    pub fn transfer(e: &Env, from: &Address, to: &Address, amount: i128) {
+        // Check if both addresses are allowed
+        if !AllowList::allowed(e, from) {
+            panic_with_error!(e, FungibleTokenError::UserNotAllowed);
+        }
+        if !AllowList::allowed(e, to) {
+            panic_with_error!(e, FungibleTokenError::UserNotAllowed);
+        }
```

Comment:
```suggestion
        if !AllowList::allowed(e, from) || !AllowList::allowed(e, to) {
            panic_with_error!(e, FungibleTokenError::UserNotAllowed);
        }
```

Reply:
seeing this as pattern in all functions below, is there a particular reason to list them one by one?